### PR TITLE
Added LM1B model implementation

### PR DESCRIPTION
**Pull request by shehadak**
This PR contains:
- Implementation of the LM1B model with support for two behavioral tasks: `next_word` and `reading_times` and the `language_system` recording target. 
- Two unit tests for each of the behavioral tasks and one for neural recording. The format of test cases follows the unit tests of GPT.

Sample run on the Perriera benchmark:
```
python brainscore_language score --model_identifier='lm1b' --benchmark_identifier='Pereira2018.243sentences-linear'
```

Out:
```
<xarray.Score ()>
array(0.58106036)
Attributes:
    raw:                   <xarray.Score ()>\narray(0.20557292)\nAttributes:\...
    ceiling:               <xarray.Score 'data' ()>\narray(0.35378928)\nAttri...
    model_identifier:      lm1b
    benchmark_identifier:  Pereira2018.243sentences-linear
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -149,3 +149,4 @@ cache
 .cache
 .idea/
 wandb/
+**/resources
```
Comment (mschrimpf) [inline .gitignore:152]: not the biggest fan of adding this broad a filter to `.gitignore`, e.g. what if some data plugin includes `data/<plugin>/resources/download_data.py` that would be automatically excluded. If it's okay with you, let's remove (or make it more specific).
```suggestion
```

File: brainscore_language/models/lm1b/__init__.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
+import functools
+import logging
+import urllib.request
+from typing import Tuple, Union, List, Dict
+
+import os
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3' # Disable verbose TF C++ output
```
Comment (mschrimpf) [inline brainscore_language/models/lm1b/__init__.py:7]:
```suggestion
os.environ['TF_CPP_MIN_LOG_LEVEL'] = os.getenv('TF_CPP_MIN_LOG_LEVEL', '3')  # Disable verbose TF C++ output
```
Allow user to set variable manually without overwriting it

File: brainscore_language/models/lm1b/__init__.py [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+                                                          for (recording_target, recording_type, layer), values
+                                                          in layer_representations.items()])),
+            'neuron_number_in_layer': ('neuroid', np.concatenate(
+                [np.arange(values.shape[-1]) for values in layer_representations.values()])),
+        }
+        neuroid_coords['neuroid_id'] = 'neuroid', functools.reduce(defchararray.add, [
+            neuroid_coords['layer'][1], '--', neuroid_coords['neuron_number_in_layer'][1].astype(str)])
+        
+        representations = NeuroidAssembly(
+            representation_values, # 1024
```
Comment (mschrimpf) [inline brainscore_language/models/lm1b/__init__.py:234]:
```suggestion
            representation_values,
```
not sure what the comment is about

File: brainscore_language/models/lm1b/__init__.py [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+        # Downloading resources (checkpoint) if not available
+        cur_dirname = os.path.dirname(os.path.realpath(__file__))
+        resources_dir = os.path.join(cur_dirname, "resources")
+        if not os.path.isdir(resources_dir):
+            os.mkdir(resources_dir)
+            downloading_iter = tqdm(RESOURCES, desc='Downloading resources for LM1B.')
+            for res in downloading_iter:
+                file_name = res.split("/")[-1]
+                file_path = os.path.join(resources_dir, file_name)
```
Comment (mschrimpf) [inline brainscore_language/models/lm1b/__init__.py:250]: suggest using pathlib
```suggestion
        cur_dirname = Path(__file__).parent
        resources_dir = cur_dirname / "resources"
        if not resources_dir.is_dir():
            resources_dir.mkdir()
            downloading_iter = tqdm(RESOURCES, desc='Downloading resources for LM1B.')
            for resource in downloading_iter:
                file_name = resource.name
                file_path = resources_dir / file_name
```

Also I would move all the downloading into its own method.

File: brainscore_language/models/lm1b/data_utils.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
```
Comment (mschrimpf) [inline brainscore_language/models/lm1b/data_utils.py:1]: if this is copied from TensorFlow, I suggest indicating the github source (including the commit, i.e. press `y` in github before copying the url)

File: brainscore_language/models/lm1b/data_utils.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+"""A library for loading 1B word benchmark dataset."""
+
+import numpy as np
+import tensorflow.compat.v1 as tf
+
+
+class Vocabulary(object):
```
Comment (mschrimpf) [inline brainscore_language/models/lm1b/data_utils.py:8]:
```suggestion
class Vocabulary:
```

**Review by mschrimpf — APPROVED** [review]
Looks good to me assuming tests pass. I suggested some changes to improve code readability. 

It looks like the `LM1B` class could actually be made into a fairly generic Tensorflow wrapper -- would that be much more work?

File: brainscore_language/models/lm1b/data_utils.py (hunk 5 above)
Comment (shehadak) [reply brainscore_language/models/lm1b/data_utils.py]: Resolved in the latest commit

File: .gitignore (hunk 1 above)
Comment (shehadak) [reply .gitignore]: Resolved in the latest commit ([f631e5b](https://github.com/brain-score/language/pull/88/commits/f631e5b86364d9ef4d58539baa52ae6f03664bcf))

**Comment by mschrimpf** [comment]
Tests pass for me
image

They fail on travis from going out of memory, I will mark them as `memory_intense` so they get excluded on travis.

**Review by kvfairchild — APPROVED** [review]
Looks good to me! Minor note: would be nice (but not necessary) to have type hinting throughout.

**Pull request merged**